ConfigurationSubstitutor/ConfigurationSubstitutor.cs

```diff
@@ -1,11 +1,17 @@
 ﻿using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Text.RegularExpressions;
 
 namespace ConfigurationSubstitution
 {
     public class ConfigurationSubstitutor
     {
+        // A shared thread static to avoid allocation on each request.
+        [ThreadStatic]
+        private static HashSet<string> inprogressCache;
```

Comment:
let's create it with the field initializer: = new HashSet...
can we rename it to _recursionDetectionSet ?

Reply:
Renamed but a ThreadStatic can't be initialised with a field initializer unfortunately.

Reply:
Oh I learned two things today! Wasn't even aware of ThreadStatic before I saw it, very smart usage of it to avoid allocations :)